modules/paapiForGpt.js

```diff
@@ -3,26 +3,45 @@
  */
 import {getHook, submodule} from '../src/hook.js';
 import {deepAccess, logInfo, logWarn, sizeTupleToSizeString} from '../src/utils.js';
-import {getGptSlotForAdUnitCode} from '../libraries/gptUtils/gptUtils.js';
 import {config} from '../src/config.js';
 import {getGlobal} from '../src/prebidGlobal.js';
 
 import {keyCompare} from '../src/utils/reducers.js';
+import {getGPTSlotsForAdUnits, targeting} from '../src/targeting.js';
+
 const MODULE = 'paapiForGpt';
 
 let getPAAPIConfig;
-
-let autoconfig = false;
+let expandFilters;
+let configWithTargeting = false;
 
 config.getConfig('paapi', (cfg) => {
-  autoconfig = deepAccess(cfg, 'paapi.gpt.autoconfig', false);
+  const cwt = deepAccess(cfg, 'paapi.gpt.configWithTargeting', false);
+  if (cwt && !configWithTargeting) {
+    logInfo(MODULE, 'enabling PAAPI configuration with setTargetingForGPTAsync')
+  }
+  configWithTargeting = cwt;
 });
 
+export function setTargetingHookFactory(getPAAPI = getPAAPIConfig, setAuctionConfigs = setComponentAuctions, expand = expandFilters) {
+  function getAuctionConfigs(adUnitCode) {
+    return getPAAPI(adUnitCode == null ? undefined : {adUnitCode}, true)
+  }
+  function getResetConfig(adUnitCode) {
+    return Object.fromEntries((adUnitCode == null ? Object.keys(expand()) : [adUnitCode]).map(au => [au, null]))
+  }
+  return function (next, adUnit, customSlotMatching) {
+    const adUnitCodes = Array.isArray(adUnit) ? adUnit : [adUnit];
+    const auctionConfigs = Object.assign({}, ...adUnitCodes.map(configWithTargeting ? getAuctionConfigs : getResetConfig))
```

Comment:
@dgirardi `setTargetingHookFactory` is called after `setPAAPIConfigForGPT`. With `configWithTargeting=false` it also removes the initially set configs from the GPT slot.

Reply:
good catch. I am not sure that there's a good fix for it. the best I can think of is to use GPT events and clear immediately after a slot is rendered - but I'd rather not go down that route.